File: wvjsb/src/main/java/com/retriable/wvjsb/Handler.java
package com.retriable.wvjsb;

import com.retriable.wvjsb.Functions.Function0;
import com.retriable.wvjsb.Functions.Function1Void;
import com.retriable.wvjsb.Functions.Function2Void;
import com.retriable.wvjsb.Functions.Function3;

import org.jetbrains.annotations.NotNull;

public final class Handler {
    /**
     *  handle event
     * @param  event event handler
     * @return current handler
     */
    public Handler onEvent(@NotNull Function3<Connection,Object,Function0<Function2Void<Object,Throwable>>,Object> event){
        this.event=event;
        return this;
    }

    /**
     * handle cancel
     * @param cancel cancel handler
     */
    public void onCancel(@NotNull Function1Void<Object> cancel){
        this.cancel=cancel;
    }

    Handler(){super();}

    Function3<Connection,Object, Function0<Function2Void<Object,Throwable>>,Object> event;
    Function1Void<Object> cancel;
}

File: wvjsb/src/main/java/com/retriable/wvjsb/Handler.kt
package com.retriable.wvjsb

class Handler {

    fun onEvent(closure:((connection: Connection,parameter: Any?,done:()->((result:Any?,t:Throwable?)->Unit))->Any?)){
        this.eventClosure=closure
    }

    fun onCancel(closure:(context:Any?)->Unit){
        this.cancelClosure=closure
    }

    internal var eventClosure:((connection: Connection, parameter: Any?, done:()->((result:Any?, t:Throwable?)->Unit))->Any?)?=null
    internal var cancelClosure:((context:Any?)->Unit)?=null
}
